src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -63,149 +63,143 @@ public class HttpWebRequestMessage : DataServiceClientRequestMessage
+                return _requestUrl;
             }
-
             set
             {
                 throw new NotSupportedException();
             }
         }
 
-        /// <summary> Returns the method for this request. </summary>
+        /// <summary>
+        /// Gets or sets the method for this request.
+        /// </summary>
         public override string Method
         {
             get
             {
-                return this.effectiveHttpMethod;
+                return _effectiveHttpMethod;
             }
-
             set
             {
                 throw new NotSupportedException();
             }
         }
 
-        /// <summary> Returns the collection of request headers. </summary>
-        public override IEnumerable<KeyValuePair<string, string>> Headers
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
         {
             get
             {
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.
         /// </summary>
         public override int Timeout
         {
-            get { return this.httpRequest.Timeout; }
-            set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets the read and write timeout (in seconds) for this request.
+        /// Gets or sets the read write timeout (in seconds) for this request.

```

Comment:
You should use sentence case here (proper punctuation and each sentence starting with an upper case letter) since this docstring will be available to the public

Reply:
In addition, since the property doesn't differ with `Timeout` when using `HttpClient` internally, you should mark it with Obsolete attribute so it can be dropped in the next major release opportunity.

Reply:
The property name is read write not read and write.

Reply:
@ElizabethOkerio Not sure I understand your response here. Can you elaborate?